src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -152,6 +152,11 @@ private bool ArgumentsEquals(Type[] genericArguments)
 
         public override int GetHashCode() => HashCode.Combine(_implementation, _type, ((System.Collections.IStructuralEquatable)Arguments).GetHashCode(EqualityComparer<CSharpType>.Default));
 
+        public CSharpType GetGenericTypeDefinition()
```

Comment:
Can you add a unit test which does what we discussed in chat?

CSharpType.Arguments should do what System.Type.GetGenericArguments() does
csharpType.GetGenericTypeDefinition().Arguments should do what type.GetGenericTypeDefinition().GetGenericArguments()

Then validate this with `new CSharpType(typeof(List<>), typeof(string))` and `new CSharpType(typeof<List<string>>)`

Reply:
Done